python/ray/serve/_private/router.py

```diff
@@ -344,7 +344,14 @@ def __init__(
         # Updated via `update_replicas`.
         self._replica_id_set: Set[str] = set()
         self._replicas: Dict[str, ReplicaWrapper] = {}
-        self._replicas_updated_event = asyncio.Event()
+
+        # NOTE(edoakes): Python 3.10 removed the `loop` parameter to `asyncio.Event`.
+        # Now, the `asyncio.Event` will call `get_running_loop` in its constructor to
+        # determine the loop to attach to. This class can be constructed for the handle
+        # from a different loop than it uses for scheduling, so we need to construct it
+        # lazily to avoid an error due to the event being attached to the wrong loop.
+        self._lazily_initialied_replicas_updated_event: Optional[asyncio.Event] = None
```

Comment:
```suggestion
        self._lazily_initialized_replicas_updated_event: Optional[asyncio.Event] = None
```

Reply:
Was theory-crafting how to add a test, just did, PTAL!